frontend/components/dashboardTile.js

```diff
@@ -12,10 +13,31 @@ var DashboardTile = React.createClass({
     },
 
     onTileClicked: function() {
-        if (this.props.tile._shouldOpenNewWindow) {
-            return openNewWindowWithLink(this.props.tile._link);
+        if (this.props.tile._linkType == 'externalLink') {
+            if (this.props.tile._shouldOpenNewWindow) {
+                return openNewWindowWithLink(this.props.tile._link);
+            } else {
+                return window.location = this.props.tile._link;
+            }
+        } else if (this.props.tile._linkType == 'resource') {
+            if (this.props.tile._shouldOpenNewWindow) {
+                return openNewWindowWithLink(this.props.tile._resourceUpload);
+            } else {
+                return window.location = this.props.tile._resourceUpload;
+            }
+        } else if (this.props.tile._linkType == 'course') {
+            var courseSlugOrId = getSlugOrId(_.find(this.props.courses, {_id: this.props.tile._courseLink}));
```

Comment:
I wonder - as a course can now be hidden from the dashboard but put in a dashboard tile - will this not find the course...

One suggestion would be to populate the _courseLink with the course model. Then you just pass in 

var courseSlugOrId = getSlugOrId(this.props.tile._courseLink)

Reply:
Maybe something useful would to be to have a little label on the coursesSelection field, where you add a course into a dashboard tile... displaying "Course is hidden from main content areas". That'll be a check for this prop `_shouldHideCourseInContentArea` from the course model.

This will avoid users getting confused as to why their dashboard tile isn't opening anything when they've added a course.